### DISCUSSION TITLE
Multi-Site: volatile sessions in Infinispan cluster

### DISCUSSION THREAD

**Discussion by tormodatt** [keycloak/keycloak#33745] [Feedback]
To follow up on https://github.com/keycloak/keycloak/pull/32452, could you give some instructions on how to test out this new feature @pruivo ?

Related epic: https://github.com/keycloak/keycloak/issues/33658

We do use Openshift actually, including the Keycloak Operator. We planned to have the Operator run two Keycloak clusters where they shared 1 Infinispan Cluster (set up with the Infinispan Operator), or 2 that are synced like in your multi-site example here: https://www.keycloak.org/high-availability/concepts-multi-site#_deployment_data_storage_and_caching if that is required

**Comment by pruivo** [comment]
Using 2 Infinispan clusters is simpler to explain because the process is documented :) 

You can follow the procedures here https://www.keycloak.org/high-availability/deploy-infinispan-kubernetes-crossdc

The only difference is the step *6. Creating the caches for Keycloak*, where you need to enable indexing in some caches for performance reasons. Let me drop the Cache CR from my cluster:

https://gist.github.com/pruivo/1e005ed9fe2351d49417e7ddf6c5e848

From the file above, you need to change:
* The namespace,
* The remote site (callout 3). It says `gh-keycloak-b` and it needs to be your site's name (ps, nothing prevents you from naming your sites gh-keycloak-a and gh-keycloak-b). 

The file is different between clusters. The file used in cluster "A" must have the cluster "B" as callout 3 and vice-versa.

**Comment by pruivo** [comment]
### Keycloak CR configuration

The procedure is here: https://www.keycloak.org/high-availability/deploy-keycloak-kubernetes

The only change is to enable the `clusterless` feature and disable the `persistent-user-sessions` feature. At the end, the CR should have the features as below:

```yaml
  features:
    enabled:
      - multi-site # <3>
      - clusterless
    disabled:
      - persistent-user-sessions
```
